test/testpy/__init__.py

```diff
@@ -48,16 +48,22 @@ def __init__(self, path, file, mode, context, config, additional=[]):
     self.mode = mode
     self.tmpdir = join(dirname(self.config.root), 'tmp')
     self.additional_flags = additional
+    if "NODE_PIPE_DIR" in os.environ:
+      self.pipeTmpDir = join(os.environ["NODE_PIPE_DIR"], 'NodePipeTmp')
```

Comment:
`"NODE_PIPE_DIR"` is still hardcoded in several places, the approach suggested in [my initial comment](https://github.com/joyent/node/pull/9381/files#r26866770) avoids that.